src/plugincollection.js

```diff
@@ -86,83 +86,114 @@ export default class PluginCollection {
 	 *  which should not be loaded (despite being specified in the `plugins` array).
 	 * @returns {Promise} A promise which gets resolved once all plugins are loaded and available into the
 	 * collection.
-	 * @returns {Array.<module:core/plugin~Plugin>} returns.loadedPlugins The array of loaded plugins.
+	 * @returns {Promise.<Array.<module:core/plugin~Plugin>>} returns.loadedPlugins The array of loaded plugins.
 	 */
 	load( plugins, removePlugins = [] ) {
 		const that = this;
 		const editor = this._editor;
 		const loading = new Set();
 		const loaded = [];
 
-		// Plugins that will be removed can be the constructors or names.
-		// We need to unify this because we are supporting loading plugins using both types.
-		removePlugins = removePlugins.reduce( ( arr, PluginConstructorOrName ) => {
-			arr.push( PluginConstructorOrName );
-
-			if ( typeof PluginConstructorOrName == 'string' ) {
-				arr.push( getPluginConstructor( PluginConstructorOrName ) );
-			} else if ( PluginConstructorOrName.pluginName ) {
-				arr.push( PluginConstructorOrName.pluginName );
-			}
+		// In order to avoid using plugin names or constructors alternatively, we map all passed plugin as plugins constructors.
+		const pluginConstructorsToLoad = plugins
+			.map( ( item ) => getPluginConstructor( item ) )
+			.filter( ( PluginConstructor ) => !!PluginConstructor );
+
+		// Check whether a length of passed array with plugins is the same like an array with mapped plugins.
+		if ( pluginConstructorsToLoad.length !== plugins.length ) {
+			// Extract names of plugins which cannot be loaded.
+			const notFoundPlugins = plugins.filter( ( item ) => !pluginConstructorsToLoad.includes( item ) );
```

Comment:
I think that this is a wrong way of getting missing plugins. In `plugins` you may have strings and in `pluginConstructorsToLoad` you will have constructors. So checking if the former is in the latter doesn't make sense. Use `getPluginConstructor()` again.